### PR TITLE
Add ZFMuonium fit function

### PR DESCRIPTION
**Pull request by hankwustfc**
Description of work:

Add ZFMuonium to Mantid fit functions.

To test:

1.Test if the documentation is added
2.Test if the fit function is working
3.Test the function with the unit test file

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by hankwustfc** [comment]
Code should work but cppcheck would not pass.

**Comment by DavidFair** [comment]
Hi @hankwustfc thanks for contributing,

The reason that cpptest isn't working is CMake, our system for helping us get all the files to the right locations whilst building Mantid, can't find `ZFMuonium.

I can either fix this for you, which will require you to run `git pull origin ZFMuonium` locally afterwards

Alternatively, you could fix this yourself by doing the following:
In `Framework/PythonInterface/test/python/plugins/functions/CMakeLists.txt` change *ZFMuonium.py* -> *ZFMuoniumTest.py*

Feel free to leave a comment if you run into any other problems

**Comment by hankwustfc** [comment]
Ahh of course. I should have noticed that. Thanks!

**Comment by DavidFair** [comment]
Ah one last hitch, there's a merge conflict with master - as someone else has updated the `CMakeLists.txt` too, so this can cause some weird failures with our tests.

Do you feel comfortable resolving it locally, or want me to quickly merge master in?

**Comment by hankwustfc** [comment]
Sure please merge master in

**Comment by DavidFair** [comment]
Okay the merge conflicts should be resolved

On your machine you'll want to pull the changes now, just to save you hassle later if you want to make any more changes

**Comment by hankwustfc** [comment]
Ok have done that. Thanks David.

**Comment by DavidFair** [comment]
@hankwustfc - Do you have examples of datasets I can test this function with?

I'm happy to accept a run number / script, both, or something else

File: Framework/PythonInterface/test/python/plugins/functions/ZFMuoniumTest.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		input = [0.0, 4.0, 8.0, 12.0]
+		expected = [0.15163664420855247, -0.05429533841911648, -0.021522983685159713, -0.02152298368515976]
+		tolerance = 1.0e-05
+		status, output = check_output("ZFMuonium", input, expected, tolerance, A0 = 0.2, FreqA = 0.3, FreqD = 0.2, FCut = 1.0,Phi = 0.0)
+		if not status:
+			msg = 'Computed output {} from input {} unequal to expected: {}'
+			self.fail(msg.format(*[str(i) for i in (output, input, expected)]))
+
+	def test_do_fit(self):
+		do_a_fit(np.arange(0.1, 16, 0.2), 'ZFMuonium', guess = dict(A0 = 0.25, FreqA = 0.35, FreqD = 0.25, FCut = 1.05,Phi = 0.0), target = dict(A0 = 0.2, FreqA = 0.3, FreqD = 0.2, FCut = 1.0,Phi = 0.0), atol = 0.01)
```
[the remark this answers is not in the document]
Comment (hankwustfc) [reply Framework/PythonInterface/test/python/plugins/functions/ZFMuoniumTest.py]: Yea sure

**Comment by AnthonyLim23** [comment]
add to release notes in #26827

**Review by NickDraper — APPROVED** [review]
builds, tests AND DOCS OK

**Pull request merged**